src/edu/ucsb/nceas/metacat/dataone/resourcemap/ResourceMapModifier.java

```diff
@@ -0,0 +1,446 @@
+/**
+ *  '$RCSfile$'
+ *  Copyright: 2000-2019 Regents of the University of California and the
+ *              National Center for Ecological Analysis and Synthesis
+ *              
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+package edu.ucsb.nceas.metacat.dataone.resourcemap;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.TimeZone;
+import java.util.Vector;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.dataone.service.types.v1.Identifier;
+import org.dataone.service.types.v1.Subject;
+import org.dataone.vocabulary.CITO;
+import org.dataone.vocabulary.DC_TERMS;
+
+import com.hp.hpl.jena.datatypes.xsd.XSDDatatype;
+import com.hp.hpl.jena.rdf.model.AnonId;
+import com.hp.hpl.jena.rdf.model.Literal;
+import com.hp.hpl.jena.rdf.model.Model;
+import com.hp.hpl.jena.rdf.model.ModelFactory;
+import com.hp.hpl.jena.rdf.model.Property;
+import com.hp.hpl.jena.rdf.model.RDFNode;
+import com.hp.hpl.jena.rdf.model.Resource;
+import com.hp.hpl.jena.rdf.model.ResourceFactory;
+import com.hp.hpl.jena.rdf.model.Selector;
+import com.hp.hpl.jena.rdf.model.SimpleSelector;
+import com.hp.hpl.jena.rdf.model.Statement;
+import com.hp.hpl.jena.rdf.model.StmtIterator;
+
+import edu.ucsb.nceas.metacat.properties.PropertyService;
+
+/**
+ * This class will create a new resource map by modifying a given resourceMap input stream. 
+ * @author tao
```

Comment:
@taojing2002 - This modifier class looks great.  I reviewed the logic, and it looks like you are changing all of the appropriate statements (`documents`/`isDocumentedBy`, `aggregates`/`isAggregatedBy`, `describes`/`isDescribedBy`).  The one question I have is about the `dcterms:identifier` statements.  Can you confirm that the old obsoleted resources are removed, which also removes the statements that include the old `dcterms:identifier` literal, like `terms:identifier>urn:uuid:326e21d5-c961-46ed-a85c-28eeedd980de</terms:identifier>`? I didn't see that in the code, by might have missed it.

Reply:
I hesitate to remove those old obsoleted resources since they may be on some triples we don't understander. Otherwise those triple may point to an orphan.  
Based on the strategy from this ticket - https://github.com/NCEAS/metacat/issues/1366, we only modify those triples we understand ( `documents/isDocumentedBy`, `aggregates/isAggregatedBy`) and reserve the all others. So I also keep those obsoleted resource in case they exist on some triples.

Reply:
I see what you mean @taojing2002 - That certainly is an issue, but I think we also need to remove all unused triples so they don't accumulate and make the resource map harder to interpret or even validate.  So I think we might need to query the Jena model and check for `subject` or `object` instances with the `identifier` literal or its `URI` equivalents (URL encoded or not).  If the identifier is used in new triples, then yes, we keep them. But if not, we remove them.  We probably should also try to validate the resulting ORE by re-parsing it with Jena and see if it throws any errors due to orphan nodes.  Let's discuss this with @gothub and @laurenwalker since they've both worked on RDF mofification in MetacatUI - we do the same operations there too.